src/test/java/org/homeschoolpebt/app/cli/TransmitterCommandsTest.java

```diff
@@ -0,0 +1,110 @@
+import java.util.Map;
+import org.homeschoolpebt.app.data.Transmission;
+import org.homeschoolpebt.app.data.TransmissionRepository;
+import org.homeschoolpebt.app.upload.CloudFile;
+import org.homeschoolpebt.app.upload.ReadOnlyCloudFileRepository;
+import org.homeschoolpebt.app.upload.S3ReadOnlyCloudFileRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("test")
+@SpringBootTest
+class TransmitterCommandsTest {
+
+  @Autowired
+  TransmitterCommands transmitterCommands;
+
+  @Autowired
+  SubmissionRepository submissionRepository;
+
+  @Autowired
+  TransmissionRepository transmissionRepository;
+
+  @Autowired
+  UserFileRepository userFileRepository;
+
+  @MockBean
+  PdfService pdfService;
+
+  @MockBean
+  ReadOnlyCloudFileRepository fileRepository;
+
+
+  Submission submission;
+
+  @BeforeEach
+  void setup() {
+    submission = Submission.builder()
+      .submittedAt(now())
+      .flow("pebt")
+      .urlParams(new HashMap<>())
+      .inputData(Map.ofEntries(
+        Map.entry("firstName", "Tester"),
+        Map.entry("lastName", "McTest")
+      )).build();
+    submissionRepository.save(submission);
+    Transmission transmission = Transmission.fromSubmission(submission);
+    transmission.setConfirmationNumber("1001");
+    transmissionRepository.save(transmission);
+
+    var submission2 = Submission.builder()
+      .submittedAt(now())
+      .flow("pebt")
+      .urlParams(new HashMap<>())
+      .inputData(Map.ofEntries(
+        Map.entry("firstName", "Other"),
+        Map.entry("lastName", "McOtherson")
+      )).build();
+    submissionRepository.save(submission2);
+
+    transmission = Transmission.fromSubmission(submission2);
+    transmission.setConfirmationNumber("1002");
+    transmissionRepository.save(transmission);
+
+    UserFile docfile = new UserFile();
+    docfile.setFilesize(10.0f);
+    docfile.setSubmission_id(submission2);
+    docfile.setOriginalName("originalFilename.png");
+    userFileRepository.save(docfile);
+  }
+
+  @Test
+  void transmitZipFile() throws IOException {
+    when(pdfService.getFilledOutPDF(any())).thenReturn("some bytes".getBytes());
+    when(pdfService.generatePdfName(any())).thenReturn("applicant_summary.pdf");
+
+    File docFile = new File("paystub.png");
+    docFile.createNewFile();
+
+    when(fileRepository.download(any())).thenReturn(new CloudFile(10L, docFile));
+
+    transmitterCommands.transmit();
+
+    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    LocalDateTime now = LocalDateTime.now();
+    String date = dtf.format(now);
+    File zipFile = new File("Apps__" + date + "__1001-1002.zip");
+    assertTrue(zipFile.exists());
+  }
+}
```

Comment:
Thanks for the tests. Do they need to clean up that ZIP file or anything?

Reply:
Yeah, it could. It's such a tiny file that I didnt both, but I'll go ahead and add a cleanup